allennlp/data/dataset_readers/textcat.py

```diff
@@ -0,0 +1,129 @@
+                 segment_sentences: bool = False,
+                 sequence_length: int = None,
+                 debug: bool = False,
+                 lazy: bool = False) -> None:
+        super().__init__(lazy=lazy)
+        self.debug = debug
+        self._word_tokenizer = word_tokenizer or WordTokenizer()
+        self._segment_sentences = segment_sentences
+        self._sequence_length = sequence_length
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+        if self._segment_sentences:
+            self._sentence_segmenter = SpacySentenceSplitter()
+
+    @overrides
+    def _read(self, file_path):
+        with open(cached_path(file_path), "r") as data_file:
+            if self.debug:
+                lines = np.random.choice(data_file.readlines(), 100)
+            else:
+                lines = data_file.readlines()
+        for line in lines:
+            if not line:
+                continue
+            items = json.loads(line)
+            text = items["text"]
+            label = str(items["label"])
+            instance = self.text_to_instance(text=text, label=label)
+            if instance is not None:
+                yield instance
+
+    @overrides
+    def text_to_instance(self, text: str, label: str = None) -> Instance:  # type: ignore
+        """
+
+        Parameters
+        ----------
+        text : ``str``, required.
+            The text to classify
+        label ``str``, optional, (default = None).
+            The label for this text.
+
+        Returns
+        -------
+        An ``Instance`` containing the following fields:
+            tokens : ``TextField``
+                The tokens in the sentence or phrase.
+            label : ``LabelField``
+                The label label of the sentence or phrase.
+        """
+        # pylint: disable=arguments-differ
+        fields: Dict[str, Field] = {}
+        text_fields: List[Field] = []
+        if self._segment_sentences:
+            sentence_tokens = self._sentence_segmenter.split_sentences(text)
+            if not sentence_tokens:
+                return None
+            for sentence in sentence_tokens:
+                word_tokens = self._word_tokenizer.tokenize(sentence)
+                if self._sequence_length is not None:
+                    if len(word_tokens) > self._sequence_length:
+                        word_tokens = word_tokens[:self._sequence_length]
+                    else:
+                        padding = [Token("@@PADDING")] * (self._sequence_length - len(word_tokens))
+                        word_tokens = word_tokens + padding
+                text_fields.append(TextField(word_tokens, self._token_indexers))
+            fields['tokens'] = ListField(text_fields)
+        else:
+            tokens = self._word_tokenizer.tokenize(text)
+            if not tokens:
+                return None
+            if self._sequence_length is not None:
```

Comment:
This entire block of code to handle `self._sequence_length` is duplicated from above, would be remove the duplication.